modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java

```diff
@@ -633,6 +633,23 @@ public String toDefaultValue(Schema p) {
         return null;
     }
 
+    @Override
+    public String toRegularExpression(String pattern) {
+        if (StringUtils.isEmpty(pattern)) {
+            return pattern;
+        }
+
+        String regexString = pattern;
+        if (!regexString.startsWith("/")) {
+            regexString = "/" + regexString;
+        }
+        if (StringUtils.countMatches(regexString, '/') == 1) {
```

Comment:
What if `regexString` contains `/`, for instance `^\d{4}\/\d{2}$`? It adds the slash at the beginning but the trailing slash won't be added.

Reply:
This is a little tricky. In your example, we would definitely want to add a `/` at the beginning and end of the `regexString`. According to the [OpenAPI Spec 3.0](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#properties), the `pattern` property should be a valid [ECMA 262 regular expression](https://www.ecma-international.org/ecma-262/5.1/#sec-7.8.5). This allows for the regular expression to define flags after the final `/`. So for a regex such as `^hello.*/i` we would only want to add the `/` at the beginning (i.e. `/^hello.*/i`). 

I couldn't find anything in the ECMA script about the available flags, but I would assume they are the same as those defined in [Mozilla's developer docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Advanced_searching_with_flags_2). If that's the case, maybe we add a check to see if the characters after the final `/` are all valid regex flags. If they are, we don't append a `/`, otherwise we do. Thoughts?

Reply:
Later in the ECMA 262 doc it explains that RegExp Objects are modeled on regular expressions in Perl 5 (https://www.ecma-international.org/ecma-262/5.1/#sec-15.10) and the Perl 5 doc talks about modifiers i.e. `/pattern/i` (https://perldoc.perl.org/perlre.html#Modifiers)

Reply:
@Raznic in [PR 1521](https://github.com/OpenAPITools/openapi-generator/pull/1521/files) @meganemura replaced the code I put in the `AbstractRubyCodegen.java` with the method `addRegularExpressionDelimiter` from `DefaultCodegen.java`.  See [code](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L3994). If you start your pattern with a forward slash it leaves it alone and if you didn't start it with a forward slash it escapes any it finds in the pattern and then surrounds the text with forward slashes.

Reply:
@ggershoni thanks for the links! I did try using `addRegularExpressionDelimiter` from `DefaultCodegen.java` initially. However, I found this method escapes any `/` in the pattern, which still results in an incorrect regex string. I see that the PR you referenced removes a few test cases where regex patterns have either a leading or trailing `/`, but not both. If we don't have to worry about those cases, then I'm okay with just checking for the leading `/` and encapsulating the pattern with `/` if necessary.

Reply:
Thanks. I wasn't aware that modifiers can be added here. It's more complicated because Python `re` module doesn't support it, it uses [flags](https://docs.python.org/3/library/re.html#module-contents) instead.

case-insensitive pattern matching - perl version:
```perl
print 'A' =~ m/a/i;    # output: 1
```

python:
```python
print(re.match(r'A', 'a', re.I))  # output:<_sre.SRE_Match object at 0x7fbb97b6d6b0>
```

We have to map Perl's modifiers to Python's flags. But for time being we can raise an exception if modifiers are used. Escaped slash (`\/`) should be treated as a normal char.

WDYT?

Reply:
I like the idea of raising an error for modifiers for the time being, though it would be prudent to capture that as a feature idea in a separate issue. And I agree with the idea of treating escaped forwarded slashes as normal characters. Time permitting, I'll try to push some new code up a bit later tonight. Thanks again for all the feedback!